### PR TITLE
Lesson4

### PR DESCRIPTION
**Pull request by vadim030303**
HW lesson4

### CODE REVIEW

File: vadims/lesson4.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+#6893 - show what next number will be
+puts 68931.next 
+
+#0 - show the previous number
+puts 0.pred 
+
+
+#12 - cast it to float, then devide by 4.5 then round it up
+puts 12.to_f / 4.7 = 2553191489361702.round() = 3
```
Comment (mtkachenk0) [inline vadims/lesson4.rb:12]: divide it by 4.5

what is `= 3` in the end?

File: vadims/lesson4.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#-123 - find out if it's odd?
+puts -123.odd?  
+
+#6893 - show what next number will be
+puts 68931.next
```
Comment (mtkachenk0) [inline vadims/lesson4.rb:5]: #6893 - show what the next number will be

you show the next for 68931

File: vadims/lesson4.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+#0 - show the previous number
+puts 0.pred 
+
+
+#12 - cast it to float, then devide by 4.5 then round it up
+puts 12.to_f / 4.7 = 2553191489361702.round() = 3
+
+#28 - call modulo by 6, then multiply result by 4.7 and round it down
+puts round_down_number = ((28 % 6) * 4.7).floor
```
Comment (mtkachenk0) [inline vadims/lesson4.rb:15]: assigning to a variable right inside puts is a bad practice

File: vadims/lesson4.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+#"Brand new bicycle\r\n" - remove trailing special characters
+puts "Brand new bicycle\r\n".chop 
+
+
+#"Brand new bicycle" - using indexes extract new
+puts "Brand new bicycle"[6..8] 
+
+#"Brand new bicycle" - delete last word
+puts "Brand new bicycle".sub("bicycle","")
```
Comment (mtkachenk0) [inline vadims/lesson4.rb:25]: method arguments should be divided with commas and spaces:
```rb
"Brand new bicycle".sub("bicycle", "")
```

File: vadims/lesson4.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+#"Brand new bicycle" - check if the sctring's last word is bicycle
+puts "Brand new bicycle".split.last == "bicycle"
+
+#"Brand new bicycle" - find out what index the letter "w" has
+puts "Brand new bicycle".index("w") = 8
```
Comment (mtkachenk0) [inline vadims/lesson4.rb:31]: what is `= 8` in the end?

**Comment by jeneamocan** [comment]
You should open Pull requests from the branch with your homework into https://github.com/mtkachenk0/saltedge-courses2020 `master` branch, as you already did

File: vadims/lesson4.rb [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+#"Brand new bicycle" - check if the sctring's last word is bicycle
+puts "Brand new bicycle".split.last == "bicycle"
+
+#"Brand new bicycle" - find out what index the letter "w" has
+puts "Brand new bicycle".index("w") = 8 
+
+#"I thought I knew everything before that night" - split the string in 3 parts to get the following result: 
+#"I thought I knew ", "everything", "before that nigh
+puts "I thought I knew everything before that night".split(/(everything)/).to_s
```
Comment (mtkachenk0) [inline vadims/lesson4.rb:35]: overengineering. Keep it simple.

First of all, you don't need to coerce anything to_s for puts operator. It's done automatically.
After that, you've used regexp which is a little bit complicated, but it has done its work.

Have you heard about `#partition` method?

**Pull request closed without merging**